### PR TITLE
[Profile] Add explorer orders link to Profile page and app menu

### LINKED ISSUE #1883: [Profile] Add a link to order list in Explorer
Discussed in https://altoros.slack.com/archives/C025G521XQD/p1637056896002300

> I think it could be a nice addone if that brings you to the explorer, at least main page where they input their wallet. On a similar note, unless you have activity in the panel, there's not easy way to find the explorer, maybe we should link it in an easy seeing place

So it would be nice to: 
 - add the 'View all orders' link to the profile page. Place it upper Trades and Referrals section
 - add the 'View all orders' link to the app menu

### PR DESCRIPTION
**Pull request by matextrem**
# Summary

Fixes #1883 

Added a link to Orders explorer from Profile page and App menu

Screen Shot 2021-11-28 at 21 36 49

  # To Test

1. Open profile page
- [x] Connect an account
- [x] You'll see a View all orders link 
2. Open app menu clicking on the three dots in the upper right corner
- [x] Connect an account
- [x] You'll see a View all orders link

### CODE REVIEW

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @matextrem , good job! Now a user may see his/her orders from the menu/profile page. 
However, some issues: 
1. I'd prefer hiding this empty section from the Profile page when a wallet is not connected
hide when not connected
2. Then, it would be good not to show it to a new user when he/she has not placed any order yet. It would be good to start displaying it when at least a user has 1 order in user's order lint in the Exploere
3. 'View all orders' link should navigate a user not only to the ` https://gnosis-protocol.io/address/...` , it should take into account the environment a user is working in: 
 - cowswap.dev.gnosisdev.com --> protocol-explorer.dev.gnosisdev.com/
 - cowswap.staging.gnosisdev.com --> protocol-explorer.staging.gnosisdev.com
 - barn.cowswap.exchange --> /barn.gnosis-protocol.io
 - cowswap.exchange --> gnosis-protocol.io

4. When connected to xDAI network. View all orders link navigates to the etherscan.io
 https://watch.screencastify.com/v/HOHCEiJ5HThSvzeW1J0p

Also, I have some doubts related to the UI. It seems to me that the section for the link is too big.. Could we make it a bit narrower?
image

Then, it might be goo to show 'View all orders' link at the top or at the bottom of the list to make it a bit more visible. @alfetopito , @alongoni , @biocom , WDYT?

**Comment by alongoni** [comment]
Maybe we can show the "View all orders" link here:
image

**Comment by alfetopito** [comment]
Great summary @elena-zh. Mati, whatever she said hehe

For `3`, there's is a function for that https://github.com/gnosis/cowswap/blob/161d8a71d692276326817439ed1d3816fab78bfb/src/custom/utils/explorer.ts#L27

For `4`, one could argue that we should always show mainnet orders since the affiliate program only counts them, but I agree with Elena and the expectation would be to link to the correct network, even if there's no tracking of those orders.

The same function above also takes care of that.

And I also like @alongoni 's suggestion. The only concern is how the top of the modal will look like on mobile.
Otherwise, I like the idea and would prefer it there instead of a whole row inside the modal.

File: src/custom/components/Menu/index.tsx [hunk 1]
```diff
@@ -254,6 +256,14 @@ export function Menu({ darkMode, toggleDarkMode }: MenuProps) {
             Code
           </span>
         </MenuItem>
+        {account && (
+          <MenuItem id="link" href={getExplorerLink(chainId || 1, account, ExplorerDataType.ORDERS)}>
```
Comment (alfetopito) [inline src/custom/components/Menu/index.tsx:260]: I see the confusion here.

This is not the explorer we are looking for :sweat_smile: 

I have what you need in a PR that I'm about to merge https://github.com/gnosis/cowswap/blob/55ce1d0e0544d5ca960f48bf4ed1713ac2dd2143/src/custom/utils/explorer.ts#L43

**Comment by elena-zh** [comment]
Hey @matextrem , great job!
Some cases: 
1.  I would vote for moving 'View all orders' menu item to the bottom to make it more visible
image
2.  'View all orders' menu item is still displayed when it is not displayed in a Profile page to a user with no orders placed:
remove from the men  as well

3. There is no  'View all orders'  link in the profile page when 'Last updated' date is not empty
no link when last updated

4. Also, different fonts and text colors for link and the date label look strange a bit. 
image

Could we display the link under the page header?
image
And in different lines in a mobile view
image

@alongoni , WDYT?

**Comment by alongoni** [comment]
hey @elena-zh  I was thinking to move the "Last updated" indicator next to the "Profile overview" title. In this way the "View all orders" link is more visible there. Do you think it could work better?
image
cc @alfetopito

**Comment by matextrem** [comment]
@elena-zh regsrding `1`, the thing is we are using /profile request info for showing/not showing orders link and using /trades request info for showing/not showing orders link on the app menu. It will be fixed once this [PR](https://github.com/gnosis/cowswap/pull/1900) is merged

**Comment by elena-zh** [comment]
@matextrem , #1900 PR takes into account trades in the Mainnet only. 
But 'view all orders' link should appear for all networks: XDAI, Rinkeby, if there are any orders there. 
Another question is: why does 'view all orders' link works as expected in the menu, but does not in the profile page?

**Comment by matextrem** [comment]
@elena-zh I'm using `/profile` request info for profile page and `/trades` request info for the app menu bar. So you are saying we are using the info given by `/trades` only ?

**Comment by elena-zh** [comment]
@matextrem , seems that this one 
`/api/v1/account/ < account > /orders/?limit=100&offset=0`

@alfetopito , am I correct?

**Comment by alfetopito** [comment]
Let's see, trying to address all comments in one place

1. To check whether to display the `View all orders` link

Here we are looking at orders rather than trades, so use the function [`getOrders`](https://github.com/gnosis/cowswap/blob/2fc6c81877/src/custom/api/gnosisProtocol/api.ts#L334) with `limit=1`, similar to how [`hasTrades`](https://github.com/gnosis/cowswap/blob/2fc6c8187796903002f9dc6b7d1d30c33b0c95ea/src/custom/utils/trade.ts#L173) function does it

2. Where should `View all orders` point to

To Explorer's `/address/<address>` endpoint. 
Simply use this function https://github.com/gnosis/cowswap/blob/2fc6c8187796903002f9dc6b7d1d30c33b0c95ea/src/custom/utils/explorer.ts#L43

3. `View all orders` placement on mobile

Looking great! Besides Ramiro's comments I have nothing to add

4. `View all orders` placement on desktop

I'm not too keen on swapping the position with `last updated` as suggested.
Maybe Ramiro's idea of putting it on another row addresses it better? 

I'd like to invoke @biocom 's Design powers to give his input and help us settle the matter.

**Comment by anxolin** [comment]
Great to see we are adding the links

I could see the two things in the right corner, view orders, last update (below), from the ideas is the one that seems better.

Anyways, I wanted just to mention that we have a menu problem 😅
This need a full redesign, not everything there is equally important, and it's hard to use that menu.

Maybe not part of this PR, but I wouldn't add a new link unless we redesign it to acomodate better the content.

image

**Comment by anxolin** [comment]
I will expand on my last comment regarding the menu.

I see that Discord, Twitter, Github could be just some nice icons that are located at the top or bottom of the menu
The links could be in sectors, where the About, Faq, Docs would be the highlight. Maybe even the stats, not sure.

Then the games could be in another section and presented differently.

File: src/custom/pages/Profile/index.tsx [hunk 2]
```diff
@@ -27,14 +27,16 @@ import { MouseoverTooltipContent } from 'components/Tooltip'
...
 
 export default function Profile() {
   const referralLink = useReferralLink()
   const { account, chainId } = useActiveWeb3React()
   const { profileData, isLoading, error } = useFetchProfile()
   const lastUpdated = useTimeAgo(profileData?.lastUpdated)
   const isTradesTooltipVisible = account && chainId == 1 && !!profileData?.totalTrades
-  const userHasTrades = !!profileData?.totalTrades
+  const gpOrders = useGpOrders(account)
+  const hasOrders = (gpOrders?.length || 0) > 0
```
Comment (alfetopito) [inline src/custom/pages/Profile/index.tsx:39]: Refactoring suggestion: given that you are doing this twice, you can create a hook for this similar to `hasTrades`

**Comment by elena-zh** [comment]
@anxolin , I opened a new issue for the menu reorganizing #1956

**Comment by elena-zh** [comment]
Works nicely in terms of functionality. 
In terms of the menu, I have created a separate issue for this (mentioned above).
Still we need to find the final decision in terms of the link displaying on the form.

**Comment by biocom** [comment]
I think it makes sense to put the 'view all orders' in the right corner:
Screen Shot 2021-12-03 at 17 48 42

File: src/custom/pages/Profile/index.tsx (hunk 2 above)
Comment (matextrem) [reply src/custom/pages/Profile/index.tsx:39]: The problem is `useGpOrders` is already a hook so moving it to a function similar to `hasTrades` would not be possible. I could instead fetch orders api inside a function call `hasOrders` but I would be repeating the same logic inside de hook.

Comment (alfetopito) [reply src/custom/pages/Profile/index.tsx:39]: What about a hook called `useHasTrades(account:string) => boolean | undefined` ?

No big deal, I'm ok if you don't agree

**Review by elena-zh — APPROVED** [review]
Changes LGTM!

**Review by alfetopito — APPROVED** [review]
:+1:

**Pull request merged**